tvloader/parser2v1/parse_file_test.go

```diff
@@ -885,4 +886,58 @@ func TestParser2_1FailsIfArtifactURIBeforeArtifactName(t *testing.T) {
 	}
 }
 
+func TestParser2_1FilesWithoutSpdxIdThrowError(t *testing.T) {
+	// case 1
+	// Last unpackaged file no packages in doc
+	// Last file of last package in the doc
+	tvPairs := []reader.TagValuePair{
+		{Tag: "SPDXVersion", Value: "SPDX-2.1"},
+		{Tag: "DataLicense", Value: "CC0-1.0"},
+		{Tag: "SPDXID", Value: "SPDXRef-DOCUMENT"},
+		{Tag: "FileName", Value: "f1"},
+	}
+	_, err := ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("files withoutSpdx Identifiers getting accepted")
+	}
+
+	// case 2 : The previous file (packaged or unpackaged does not contain spdxID)
+	tvPairs = append(tvPairs, reader.TagValuePair{Tag: "FileName", Value: "f2"})
+	_, err = ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("%s", err)
```

Comment:
Note that this won't work -- the test is `if err == nil`, so if we get to the `Errorf()` call then err will be nil and can't be printed. I think you want to replace this with an actual description of what we expected and what we incorrectly got instead?